### PR TITLE
Fix __init__.py when pkg_resources is missing

### PR DESCRIPTION
**Pull request by calebmadrigal**
Simply copied the `pkg_resources` fake fallback from `tldextract.py` into `__init__.py` to prevent failure if `pkg_resources` is missing.

### CODE REVIEW

File: tldextract/__init__.py [hunk 1]
```diff
@@ -1,6 +1,20 @@
...
+try:
+    import pkg_resources
+except ImportError:
+    class pkg_resources(object):  # pylint: disable=invalid-name
+
+        """Fake pkg_resources interface which falls back to getting resources
+        inside `tldextract`'s directory.
+        """
+        @classmethod
+        def resource_stream(cls, _, resource_name):
```
Comment (john-kurkowski) [inline tldextract/__init__.py]: In this file, the fake fallback is only used for `pkg_resources.get_distribution`, a few lines down (outside of the range I can comment on). This fake doesn't define that method. Does this trade an `ImportError` for an `AttributeError`?

**Comment by john-kurkowski** [comment]
Surprised this has slipped past for so long! Any thoughts how to add test coverage?

File: tldextract/__init__.py (hunk 1 above)
Comment (calebmadrigal) [inline tldextract/__init__.py]: Nice catch. We would need to define `get_distribution`. I don't know if there's a good way to get the version of the package without `pkg_resources`. Perhaps we could just add `ImportError` to the exception handler that catches `pkg_resources.DistributionNotFound` below?

**Comment by calebmadrigal** [comment]
@john-kurkowski What do you think of this alternative approach?

**Comment by john-kurkowski** [comment]
I like it! Just need to fix or work around the CI failure.

**Comment by john-kurkowski** [comment]
I deduped the `__version__` calculation. Not sure why we didn't do that before. Does that work?

**Comment by calebmadrigal** [comment]
Looks good to me!

**Comment by john-kurkowski** [comment]
🙌

**Pull request merged**